src/schedule/zephyr_domain.c

```diff
@@ -67,6 +72,16 @@ static void zephyr_domain_thread_fn(void *p1, void *p2, void *p3)
 		/* immediately go to sleep, waiting to be woken up by the timer */
 		k_sem_take(&dt->sem, K_FOREVER);
 
+#if CONFIG_CROSS_CORE_STREAM
+		if (atomic_get(&zephyr_domain->block)) {
```

Comment:
@serhiy-katsyuba-intel what's the thinking behind using an atomic_get()? I think the normal pattern with condvars is to take the mutex always, check the condition (with lock held), if condition is not met, call k_condvar_wait(), else if condition is met, release the mutex and exit.

Reply:
Ack @serhiy-katsyuba-intel . Would be good to sanity check the cost -- the uncontented case should be pretty fast for mutex_lock, just taking a spinlock and checking lock owner. The slower path is only taken if lock is held. Cost on xtensa needs to be checked though -- it's true this is a high-call rate place.

If we keep this optimization, a comment would be good. I think there's a race now:
```
core1:   if (atomic_get(&block))   // returns 0, so branch not taken
core2:     mutex_lock(&block_mutex)  
core2:     atomic_set(&block, 1)
core1:   dt->handler(dt->arg);
core2:     // continue execution after ll_block()
```

I think your code is safe as you'll send the extra IDC to complete the current LL task, but this would be good to note as a comment (that this is per design).

Reply:
> I guess taking/releasing mutex is quite expensive.

@serhiy-katsyuba-intel Best to also check with @marcinszkudlinski as he has the data showing what is cheap/expensive re locking on xtensa.

Reply:
Added comment with description before that "if".

Reply:
I think the comment should rather describe *why* checking that atomic variable without locking is enough. The reason is - as @kv2019i has pointed out - the IDC. So what we're doing is something like
```
IDC <core X>				LL <core Y>
set flag				scheduler running
wait-for-scheduler-idle-on-Y
					scheduler completes
			<SYNC POINT>
process IDC				scheduler starts but detects the flag, waits on condvar
signal condvar
			<SYNC POINT>
					scheduler proceeds running the pipeline
```

Reply:
That's a good point @lyakh @serhiy-katsyuba-intel . There's two critical pieces of info anyone reading the code needs to understand: 1) the classic condvar process is modified here on purpose (this is documented), 2) this modification is safe because of logic elsewhere (the blocking IDC).

Let's proceed with this now as we got both CI and checks. I'm happy with the (1) part as that's enough for the reviewer/maintainer to alert this is done on purpose.